### PR TITLE
8278923: Document Klass::is_loader_alive

### PR DESCRIPTION
**Pull request by coleenp**
This trivial change just adds a comment to Klass::is_loader_alive.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8278923](https://bugs.openjdk.org/browse/JDK-8278923): Document Klass::is_loader_alive

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9400/head:pull/9400` \
`$ git checkout pull/9400`

Update a local copy of the PR: \
`$ git checkout pull/9400` \
`$ git pull https://git.openjdk.org/jdk pull/9400/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9400`

View PR using the GUI difftool: \
`$ git pr show -t 9400`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9400.diff">https://git.openjdk.org/jdk/pull/9400.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/oops/klass.inline.hpp [hunk 1]
```diff
@@ -44,6 +44,10 @@ inline bool Klass::is_non_strong_hidden() const {
 // Iff the class loader (or mirror for non-strong hidden classes) is alive the
 // Klass is considered alive. This is safe to call before the CLD is marked as
 // unloading, and hence during concurrent class unloading.
+// This returns false if the Klass is unloaded, or about to be unloaded because the holder of
+// the CLD is strongly reachable.
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/klass.inline.hpp:48]: Is that a typo? I would expect being unreachable to lead to unloading?

Comment (coleenp) [reply src/hotspot/share/oops/klass.inline.hpp:48]: It is a typo thanks.

**Comment by coleenp** [comment]
Thank you David!

**Review by iklam — APPROVED** [review]
LGTM

**Pull request closed without merging**